### DISCUSSION TITLE
Use Buffers in Auto-Completion

### DISCUSSION THREAD

**Discussion by flatcap** [neomutt/neomutt#3899] [Development]
Here's the layout of functions in Auto-Completion:

<img src="https://flatcap.org/mutt/completion2.svg">

**Source**: [graphviz](https://flatcap.org/mutt/completion2.gv), [svg](https://flatcap.org/mutt/completion2.svg)

On the left, `buf_get_field()` and its 66 callers take a `struct Buffer` parameter.
Everything to the right needs converting.

<details>
<summary>Spoiler</summary>

Start by changing the `EnterWindowData` to use **both** a `Buffer` and `char*` (they'd point to the same memory).
This would allow you to convert each of the Editor's functions as a separate commit.
e.g. `<op_editor_complete>` / `op_editor_complete()`.

The `complete_*()` functions take `EnterWindowData`, so they will only need tweaking as you convert the actual completion functions, on the right.

</details>

**Comment by roccoblues** [comment]
This has been done in https://github.com/neomutt/neomutt/commit/02a97ee809acd396fa6a7c62f3530a46655c98f7 and can be closed right?

**Comment by flatcap** [comment]
Yes.
The next stage is to factor out all the functions on the right.
That's been adopted by Amudtogal in #3955 Neomutt completion API
